From ab5c83b8d55f18e583d305cb4091157044c69ecf Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@gmail.com>
Date: Wed, 20 Aug 2008 14:57:21 +0300
Subject: [PATCH] 02_wildmidi-config.patch

---
 ext/timidity/gstwildmidi.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ext/timidity/gstwildmidi.c b/ext/timidity/gstwildmidi.c
index fa5b89d..86d5a16 100644
--- a/ext/timidity/gstwildmidi.c
+++ b/ext/timidity/gstwildmidi.c
@@ -55,7 +55,7 @@
 #include "gstwildmidi.h"
 
 #ifndef WILDMIDI_CFG
-#define WILDMIDI_CFG "/etc/timidity.cfg"
+#define WILDMIDI_CFG "/etc/wildmidi/wildmidi.cfg"
 #endif
 
 GST_DEBUG_CATEGORY_STATIC (gst_wildmidi_debug);
@@ -185,7 +185,7 @@ wildmidi_open_config ()
   }
 
   if (path == NULL) {
-    path = g_build_path (G_DIR_SEPARATOR_S, "/etc", "timidity.cfg", NULL);
+    path = g_build_path (G_DIR_SEPARATOR_S, "/etc", "timidity", "timidity.cfg", NULL);
     GST_DEBUG ("trying %s", path);
     if (path && (g_access (path, R_OK) == -1)) {
       g_free (path);
