From f1229a69c8e9d81a8ff24d90551e93b207f03c9e Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Ren=C3=A9=20Stadler?= <rene.stadler@nokia.com>
Date: Mon, 23 Feb 2009 18:14:12 +0200
Subject: [PATCH] Lower gettext requirement from 0.17 to 0.13.

---
 autogen.sh   |    2 +-
 configure.ac |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 8b88b9b..7317bfd 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -40,7 +40,7 @@ version_check "autoconf" "$AUTOCONF autoconf autoconf-2.54 autoconf-2.53 autocon
 version_check "automake" "$AUTOMAKE automake automake-1.11 automake-1.10 automake-1.9 automake-1.7 automake-1.6 automake-1.5" \
               "ftp://ftp.gnu.org/pub/gnu/automake/" 1 7 || DIE=1
 version_check "autopoint" "autopoint" \
-              "ftp://ftp.gnu.org/pub/gnu/gettext/" 0 17 || DIE=1
+              "ftp://ftp.gnu.org/pub/gnu/gettext/" 0 13 || DIE=1
 version_check "libtoolize" "$LIBTOOLIZE libtoolize glibtoolize" \
               "ftp://ftp.gnu.org/pub/gnu/libtool/" 1 5 0 || DIE=1
 version_check "pkg-config" "" \
diff --git a/configure.ac b/configure.ac
index ac22aef..1680efc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,7 +58,7 @@ AC_SUBST(ACLOCAL_AMFLAGS, "-I m4 -I common/m4")
 
 dnl set up gettext
 dnl the version check needs to stay here because autopoint greps for it
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.13])
 AM_GNU_GETTEXT([external])
 AG_GST_GETTEXT([gst-plugins-bad-$GST_MAJORMINOR])
 
