From 2aafda4f236c60ecd58255a7192ecc6ceb083268 Mon Sep 17 00:00:00 2001
From: Lasse Laukkanen <ext-lasse.2.laukkanen@nokia.com>
Date: Mon, 30 Mar 2009 14:08:07 +0300
Subject: [PATCH] camerabin: update new clock when stopping video recording

---
 gst/camerabin/gstcamerabin.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index e6bb7e8..3b5ab63 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -1929,6 +1929,15 @@ gst_camerabin_reset_to_view_finder (GstCameraBin * camera)
   /* Set video bin to READY state */
   if (camera->active_bin == camera->vidbin) {
     state_ret = gst_element_set_state (camera->active_bin, GST_STATE_READY);
+
+    /* Set to PAUSED to update new clock after stopping video recording */
+    /* Should we set videobin to NULL and post clock-lost message and 
+       let application do this? */
+    if (GST_STATE (camera) == GST_STATE_PLAYING) {
+      gst_element_set_state (GST_ELEMENT (camera), GST_STATE_PAUSED);
+      gst_element_set_state (GST_ELEMENT (camera), GST_STATE_PLAYING);
+    }
+
     if (state_ret == GST_STATE_CHANGE_FAILURE) {
       GST_WARNING_OBJECT (camera, "state change failed");
       gst_element_set_state (camera->active_bin, GST_STATE_NULL);
