From 66f6cdf564af5cd08a15e699a6c001a2a906cc61 Mon Sep 17 00:00:00 2001
From: Lasse Laukkanen <ext-lasse.2.laukkanen@nokia.com>
Date: Tue, 25 Aug 2009 10:10:15 +0300
Subject: [PATCH] camerabin: fix deadlock when error occurs at video recording startup

---
 gst/camerabin/gstcamerabin.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index 4c51c0f..f0a47d2 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -1613,6 +1613,8 @@ gst_camerabin_start_video_recording (GstCameraBin * camera)
 
   if (state_ret != GST_STATE_CHANGE_FAILURE) {
     g_mutex_lock (camera->capture_mutex);
+    camera->capturing = TRUE;
+    g_mutex_unlock (camera->capture_mutex);
     gst_element_set_locked_state (camera->vidbin, FALSE);
     g_object_set (G_OBJECT (camera->src_out_sel), "resend-latest", FALSE,
         "active-pad", camera->pad_src_vid, NULL);
@@ -1622,11 +1624,8 @@ gst_camerabin_start_video_recording (GstCameraBin * camera)
             "capture-mode")) {
       g_object_set (G_OBJECT (camera->src_vid_src), "capture-mode", 2, NULL);
     }
-
     gst_element_set_state (GST_ELEMENT (camera), GST_STATE_PLAYING);
     gst_element_set_locked_state (camera->vidbin, TRUE);
-    camera->capturing = TRUE;
-    g_mutex_unlock (camera->capture_mutex);
   } else {
     GST_WARNING_OBJECT (camera, "videobin state change failed");
     gst_element_set_state (camera->vidbin, GST_STATE_NULL);
