From 8d2aebd9c1241711beee4a49aa3223fb07a27837 Mon Sep 17 00:00:00 2001
From: Lasse Laukkanen <ext-lasse.2.laukkanen@nokia.com>
Date: Wed, 11 Mar 2009 18:31:17 +0200
Subject: [PATCH] camerabin: change default yuv format from I420 to UYVY

---
 gst/camerabin/gstcamerabin.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index 09355fd..e6bb7e8 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -208,7 +208,7 @@ static guint camerabin_signals[LAST_SIGNAL];
 #define DEFAULT_FPS_N 0         /* makes it use the default */
 #define DEFAULT_FPS_D 1
 
-#define CAMERABIN_DEFAULT_VF_CAPS "video/x-raw-yuv,format=(fourcc)I420"
+#define CAMERABIN_DEFAULT_VF_CAPS "video/x-raw-yuv,format=(fourcc)UYVY"
 #define CAMERABIN_MAX_VF_WIDTH 848
 #define CAMERABIN_MAX_VF_HEIGHT 848
 
